### ISSUE TITLE
Investigate synchronization errors

### ISSUE THREAD

**Issue by jonastheis** [iotaledger/goshimmer#525]
After starting a node and letting it sync in our little test net it does not change its status to synchronized even though the MPS rate indicates that it is finished with the sync (only 2 MPS if the bootstrapping nodes are shown).

**Comment by Wollac** [comment]
I observed a drastic spike of the goroutine count when a larger chunk of messages becomes synced:
`race: limit on 8128 simultaneously alive goroutines is exceeded, dying`
Maybe it is related, but it should also be fixed.

**Comment by luca-moser** [comment]
Shrink the message layer solidifier worker pool to NumCPU()s

**Comment by lzpap** [comment]
I'm afraid https://github.com/iotaledger/hive.go/pull/144 and https://github.com/iotaledger/goshimmer/pull/549 didn't solve this issue. When I start a node on our test network, it still doesn't get synced and the `sync` plugin cannot find the solidified anchor messages.

**Comment by Wollac** [comment]
Seems like we are currently seeing two competing chains of which only can be solidified:
![Tangle](https://user-images.githubusercontent.com/4930426/85752695-eacc4400-b70b-11ea-95d1-a89fab245281.png)
If the anchors are picked from the non-solid chain this could explain this behavior.

**Comment by lzpap** [comment]
Something has happened 🎉🎉🎉
```
2020-06-25T18:20:26+02:00       INFO    Node    node/node.go:95 Loading Plugin: Sync ... done
2020-06-25T18:20:26+02:00       INFO    Sync    sync/plugin.go:185      monitoring for synchronization, awaiting 3 anchor point messages to become solid
2020-06-25T18:20:26+02:00       INFO    Node    node/node.go:117        Starting Plugin: Sync ... done
2020-06-25T18:20:48+02:00       INFO    Sync    sync/plugin.go:195      added message ExefhzijQN as anchor point (1 of 3 collected)
2020-06-25T18:20:48+02:00       INFO    Sync    sync/plugin.go:195      added message rjUScjngqH as anchor point (2 of 3 collected)
2020-06-25T18:20:48+02:00       INFO    Sync    sync/plugin.go:195      added message FECvDxmnmK as anchor point (3 of 3 collected)
2020-06-25T18:20:49+02:00       INFO    Sync    sync/plugin.go:205      anchor message FECvDxmnmK has become solid
2020-06-25T18:20:49+02:00       INFO    Sync    sync/plugin.go:205      anchor message rjUScjngqH has become solid
2020-06-25T18:20:49+02:00       INFO    Sync    sync/plugin.go:205      anchor message ExefhzijQN has become solid
2020-06-25T18:20:49+02:00       INFO    Sync    sync/plugin.go:237      all anchor messages have become solid, marking node as synced
2020-06-25T18:20:49+02:00       INFO    Sync    sync/plugin.go:113      monitoring for desynchronization

```

**Comment by Wollac** [comment]
I did some more tests and when the tangle to be synced gets large enough, even after https://github.com/iotaledger/goshimmer/pull/549 I still see
`race: limit on 8128 simultaneously alive goroutines is exceeded, dying`
The number of goroutines spawned still seems to depend on the size of the Tangle.

**Comment by wonkytonky2** [comment]
2020-06-30T22:04:52+02:00       INFO    Sync    sync/plugin.go:235      freeing anchor point slot of 41w6sWJXyj as it didn't become solid within 1m0s
2020-06-30T22:04:52+02:00       INFO    Sync    sync/plugin.go:235      freeing anchor point slot of 42gNKJnSXL as it didn't become solid within 1m0s
2020-06-30T22:04:52+02:00       INFO    Sync    sync/plugin.go:235      freeing anchor point slot of 4am5AEMiGC as it didn't become solid within 1m0s
2020-06-30T22:04:54+02:00       INFO    Sync    sync/plugin.go:201      added message 5hjNc9qJD6 as anchor point (1 of 3 collected)
2020-06-30T22:04:57+02:00       INFO    Sync    sync/plugin.go:201      added message d1Y6sgzr1b as anchor point (2 of 3 collected)
2020-06-30T22:05:00+02:00       INFO    Sync    sync/plugin.go:201      added message 28WToTttGd as anchor point (3 of 3 collected)

**Comment by Wollac** [comment]
It seems like the syncing becomes much less reliable with slower hardware and/or internet connection. While it works on reasonable VPS every time in a few minutes, syncing seems to break more often the lower hardware you use.

**Comment by c-mnd** [comment]
Fuck, I copied the db from one node to another. It always ran once, but never twice!
Might that be the difference?

**Comment by c-mnd** [comment]
ran == synced

**Comment by capossele** [comment]
Fixed by #650 and #670
